### PR TITLE
feat(ReconnectingWebSocket): Refactor package

### PR DESCRIPTION
**Pull request by inglec-arista**
- Upgrade all dependencies to latest
- Move `ReconnectEvent` to separate file
- Unify `ReconnectEvent` constructor event init args
- Export `ReconnectingWebSocketEventMap`
- Partially apply default options
- Move "RWS" debug string to `dbg` method
- Use optional chaining for `this.ws` checks
- Replace `this.ws.onx` with `this.ws.addEventListener`
- Widen `send` data type
- Fix `addEventListener` to apply to all new WebSockets
- Annotate `Options` type

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,23 +42,23 @@
...
-    "rollup": "^1.26.3",
-    "rollup-plugin-node-builtins": "^2.1.2",
-    "rollup-plugin-node-resolve": "^5.2.0",
-    "rollup-plugin-terser": "^5.1.2",
-    "rollup-plugin-typescript": "^1.0.1",
-    "tslib": "^1.10.0",
-    "typedoc": "^0.15.0",
-    "typedoc-plugin-markdown": "^2.2.11",
-    "typescript": "^3.7.2"
+    "@rollup/plugin-node-resolve": "13.3.0",
```
Comment (rufman) [inline package.json]: use caret version ranges

File: src/index.ts [hunk 2]
```diff
@@ -1,119 +1,147 @@
...
-  /** Passed `ReconnectEvent` in reality. */
-  public onerror = (event: Event): Event => event;
-  public onmessage = (event: MessageEvent): MessageEvent => event;
+  /**
+   * @deprecated Listener signature is simplified in order to correctly implement `WebSocket`.
+   * Use `addEventListener('close', listener)`.
+   */
+  public onclose: WebSocket['onclose'] = null;
+  public onconnecting:
+    | ((this: this, event: ReconnectingWebSocketEventMap['connecting']) => void)
```
Comment (rufman) [inline src/index.ts:77]: this onclose etc methods should still work, as they are valid syntax for interacting with websockets.

**Review by rufman — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (inglec-arista) [reply src/index.ts:77]: Yes, and the property methods do still work!
Maybe the `@deprecated` tag is not suitable, but I did want to note that TS clients should prefer the `addListener` method for the correct listener typing.

File: package.json (hunk 1 above)
Comment (inglec-arista) [reply package.json]: Done

File: src/index.ts (hunk 2 above)
Comment (inglec-arista) [reply src/index.ts:77]: Removed the `@deprecated` tag

**Review by rufman — APPROVED** [review]

**Pull request merged**